openmc/lib/core.py

```diff
@@ -114,6 +114,198 @@ class _SourceSite(Structure):
 _dll.openmc_get_mean_optical_thickness_between_voxels.argtypes = [_Position, _Position, _Position, _Position, c_int, POINTER(c_double)]
 _dll.openmc_calculate_optical_thickness.restype = c_double
 _dll.openmc_calculate_optical_thickness.argtypes = [_Position, _Position]
+_dll.openmc_get_voxel_cross_sections.restype = None
+_dll.openmc_get_voxel_cross_sections.argtypes = [_Position, _Position, c_int, POINTER(c_double), POINTER(c_double)]
+_dll.openmc_get_transport_operator_between_voxels.restype = None
+_dll.openmc_get_transport_operator_between_voxels.argtypes = [_Position, _Position, _Position, _Position, c_int, POINTER(c_double)]
+_dll.openmc_get_self_transport_operator.restype = None
+_dll.openmc_get_self_transport_operator.argtypes = [_Position, _Position, c_int, POINTER(c_double)]
+_dll.openmc_calculate_g_ij.restype = None
+_dll.openmc_calculate_g_ij.argtypes = [_Position, _Position, _Position, _Position, c_int, POINTER(c_double)]
+_dll.openmc_calculate_g_ii.restype = None
+_dll.openmc_calculate_g_ii.argtypes = [_Position, _Position, c_int, POINTER(c_double)]
+
+def get_voxel_cross_sections(voxel_min:tuple, voxel_max:tuple, num_samples:int):
+    """Calculate average total and absorption cross sections within a voxel.
+    
+    Parameters
+    ----------
+    voxel_min : tuple
+        Minimum corner coordinates (x, y, z) of the voxel
+    voxel_max : tuple
+        Maximum corner coordinates (x, y, z) of the voxel
+    num_samples : int
+        Number of random samples to use for averaging
+        
+    Returns
+    -------
+    tuple
+        A tuple containing (sigma_t, sigma_a) - the average total and 
+        absorption cross sections
+    """
+    min_x, min_y, min_z = voxel_min
+    max_x, max_y, max_z = voxel_max
+    
+    min_pos = _Position(c_double(min_x), c_double(min_y), c_double(min_z))
+    max_pos = _Position(c_double(max_x), c_double(max_y), c_double(max_z))
+    
+    sigma_t = c_double()
+    sigma_a = c_double()
+    
+    _dll.openmc_get_voxel_cross_sections(
+        min_pos, max_pos, c_int(num_samples), 
+        sigma_t, sigma_a
+    )
+    
+    return sigma_t.value, sigma_a.value
+
+def get_transport_operator_between_voxels(start_voxel_min:tuple, start_voxel_max:tuple, 
```

Comment:
I assume this and the next one are just for debugging - they are called by `get_g_ij` adn `get_g_ii` from the C++ side.